server/webapp/WEB-INF/rails.new/spec/javascripts/version_updater_spec.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2015 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe("VersionUpdater", function(){
+    beforeEach(function () {
+        localStorage.clear();
+    });
+
+    describe("update", function(){
+        it("should_update_the_latest_version", function(){
+            var versionInfo = {component_name: 'go_server', update_server_url: 'update_server_url'}
+            var versionUpdater = new VersionUpdater("stale_version_info_url", "update_server_version_info_url");
+
+            spyOn(versionUpdater, 'fetchLatestVersion').and.callFake(function(params){})
+            spyOn(jQuery, "ajax").and.callFake(function(options) {
+                options.success(versionInfo);
+            });
+
+            versionUpdater.update();
+
+            expect(jQuery.ajax).toHaveBeenCalled();
+            expect(versionUpdater.fetchLatestVersion).toHaveBeenCalledWith(versionInfo);
+       });
+
+        it("should_not_update_the_latest_version_if_updated_in_last_half_an_hour", function(){
+            spyOn(localStorage, 'getItem').and.callFake(function(key){
```

Comment:
Is there a need to spy on this when you can actually test what's in the localstorage?